jkube-kit/build/service/docker/src/test/java/org/eclipse/jkube/kit/build/service/docker/RegistryServicePullIT.java

```diff
@@ -0,0 +1,244 @@
+      "word.word/word.word/word:tag,/v1.18/images/create?fromImage=word.word%2Fword.word%2Fword&tag=tag",
+      "word.word.word/word:tag,/v1.18/images/create?fromImage=word.word.word%2Fword&tag=tag",
+      "word.word.word/word/word:tag,/v1.18/images/create?fromImage=word.word.word%2Fword%2Fword&tag=tag"
+  })
+  void pullImageWithPolicy_whenNoRegistryFromAnySource_thenPullImage(String imageName, String pullImageEndpoint) throws IOException {
+    // Given
+    server.expect().post()
+        .withPath(pullImageEndpoint)
+        .andReturn(200, Collections.singletonMap("message", "OK"))
+        .once();
+    RegistryService registryService = createNewRegistryService(server.getPort());
+    ImagePullManager imagePullManager = ImagePullManager.createImagePullManager("Always", "true", new Properties());
+    BuildConfiguration buildConfiguration = BuildConfiguration.builder().build();
+    RegistryConfig registryConfig = createNewRegistryConfig(null);
+
+    // When
+    registryService.pullImageWithPolicy(imageName, imagePullManager, registryConfig, buildConfiguration);
+
+    // Then
+    verify(kitLogger).info(eq("Pulled %s in %s"), eq(imageName), anyString());
+  }
+
+  @ParameterizedTest(name = "pull {0} and registry from registry config, then registry not prepended to image name {1}")
+  @CsvSource({
+      "word.word/word/word:tag,/v1.18/images/create?fromImage=word.word%2Fword%2Fword&tag=tag",
+      "word:5000/word:tag,/v1.18/images/create?fromImage=word%3A5000%2Fword&tag=tag",
+      "word.word:5000/word:tag,/v1.18/images/create?fromImage=word.word%3A5000%2Fword&tag=tag",
+      "word.word/word.word/word:tag,/v1.18/images/create?fromImage=word.word%2Fword.word%2Fword&tag=tag",
+      "word.word.word/word/word:tag,/v1.18/images/create?fromImage=word.word.word%2Fword%2Fword&tag=tag"
+  })
+  void pullImageWithPolicy_whenRegistryFromRegistryConfig_thenRegistryNotPrependedToImageName(String imageName, String pullImageEndpoint) throws IOException {
+    // Given
+    server.expect().post()
+        .withPath(pullImageEndpoint)
+        .andReturn(200, Collections.singletonMap("message", "OK"))
+        .once();
+    RegistryService registryService = createNewRegistryService(server.getPort());
+    ImagePullManager imagePullManager = ImagePullManager.createImagePullManager("Always", "true", new Properties());
+    BuildConfiguration buildConfiguration = BuildConfiguration.builder().build();
+    RegistryConfig registryConfig = createNewRegistryConfig("quay.io");
+
+    // When
+    registryService.pullImageWithPolicy(imageName, imagePullManager, registryConfig, buildConfiguration);
+
+    // Then
+    verify(kitLogger).info(eq("Pulled %s in %s"), eq(imageName), anyString());
+  }
+
+  @ParameterizedTest(name = "pull {0} and registry from registry config, then registry prepended to image name {1}")
+  @MethodSource("pullImageImageNameRegistryOverriddenWithRegistryFromRegistryConfigParams")
```

Comment:
Please, try to keep the input data method close/next to this one so that's easier to check